provider/export/src/export_impl.rs

```diff
@@ -328,90 +326,82 @@ fn select_locales_for_marker<'a>(
         return Ok(selected_locales);
     }
 
-    // The "candidate" langids that could be exported is the union of requested and supported.
-    let all_candidate_langids = supported_map
+    // The "candidate" locales that could be exported is the union of requested and supported.
+    let all_candidate_locales = supported_map
         .keys()
         .chain(requested_families.keys())
         .collect::<HashSet<_>>();
 
     // Compute a map from LanguageIdentifiers to DataLocales, including inherited auxiliary keys
     // and extensions. Also resolve the ancestors and descendants while building this map.
-    let mut selected_langids = requested_families.keys().cloned().collect::<HashSet<_>>();
-    let expansion_map: HashMap<&LanguageIdentifier, HashSet<DataIdentifierCow>> =
-        all_candidate_langids
-            .into_iter()
-            .map(|current_langid| {
-                let mut expansion = supported_map
-                    .get(current_langid)
-                    .cloned()
-                    .unwrap_or_default();
-                if include_full && !selected_langids.contains(current_langid) {
-                    log::trace!("Including {current_langid}: full locale family: {marker:?}");
-                    selected_langids.insert(current_langid.clone());
+    let mut selected_locales = requested_families.keys().cloned().collect::<HashSet<_>>();
+    let expansion_map: HashMap<&DataLocale, HashSet<DataIdentifierCow>> = all_candidate_locales
+        .into_iter()
+        .map(|current_locale| {
+            let mut expansion = supported_map
+                .get(current_locale)
+                .cloned()
+                .unwrap_or_default();
+            if include_full && !selected_locales.contains(current_locale) {
+                log::trace!("Including {current_locale}: full locale family: {marker:?}");
+                selected_locales.insert(current_locale.clone());
+            }
+            if current_locale.language().is_empty() && current_locale.is_und() {
```

Comment:
Previously this was

```rust
if current_langid.language.is_empty() && current_langid != &LanguageIdentifier::UND
```

so you are missing a `!`

```suggestion
            if current_locale.language().is_empty() && !current_locale.is_und() {
```

Reply:
omg thank you